### PR TITLE
Fix POD_NAME env

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Tested with @lec-bit 

cc @YaoZengzeng @nlgwcy

File: deploy/helm/templates/daemonset.yaml [hunk 1]
```diff
@@ -26,24 +26,31 @@ spec:
...
         - name: MESH_CONTROLLER
           value: {{ quote .Values.deploy.kmesh.env.meshController }}
         - name: BPF_LOG_SIZE
           value: {{ quote .Values.deploy.kmesh.env.bpfLogSize }}
         - name: KUBERNETES_CLUSTER_DOMAIN
           value: {{ quote .Values.kubernetesClusterDomain }}
+        - name: SERVICE_ACCOUNT
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
```
Comment (YaoZengzeng) [inline deploy/helm/templates/daemonset.yaml:47]: `apiVersion` here could also be deleted?

File: pkg/controller/config/config.go [hunk 2]
```diff
@@ -93,12 +108,37 @@ func (c *XdsConfig) Init() error {
 
 	log.Infof("service node %v connect to discovery address %v", c.ServiceNode, c.DiscoveryAddress)
 
+	c.Metadata.Namespace = podNamespace
+	c.Metadata.ClusterID = cluster.ID(clusterID)
+	c.Metadata.InstanceIPs = []string{ip}
+	// TODO: add labels to support localiy load balancing
+	c.Metadata.Labels = nil
+	c.Metadata.MeshID = "cluster.local"
```
Comment (YaoZengzeng) [inline pkg/controller/config/config.go:116]: Should `MeshID` also be allowed to get from env or command line flag?

Comment (hzxuzhonghu) [reply pkg/controller/config/config.go:116]: sgtm

File: deploy/helm/templates/daemonset.yaml (hunk 1 above)
Comment (hzxuzhonghu) [reply deploy/helm/templates/daemonset.yaml:47]: i think so, this is an optional field, and default to `v1`

**Review by hzxuzhonghu — COMMENTED** [review]
@nlgwcy have a look when you are free

**Pull request merged**